### PR TITLE
Control mode on zip inputs (#96)

### PR DESCRIPTION
**Pull request by statik**
This seems to fix the issue. Tried adding a test, I'm happy to improve/polish based on your feedback.

### CODE REVIEW

File: pkg/build_zip.py [hunk 1]
```diff
@@ -65,6 +66,12 @@ def main(args):
       dst_path = _combine_paths(args.directory, dst_path)
 
       entry_info = ZipInfo(filename=dst_path, date_time=ts)
+      st = os.stat(src_path)
```
Comment (aiuto) [inline pkg/build_zip.py]: This is not a general solution. See the comment in #96.

Comment (statik) [reply pkg/build_zip.py]: Thanks for reviewing! I've redone the change to use a mode passed in the BUILD files with a sensible default modeled after what is done in pkg_tar.

**Comment by Gloix** [comment]
Does the mode flag affect all source files? Say I want to pack an executable and a README file. Will they both have to be executable?

**Comment by mitchelldavis** [comment]
Can we get this pushed through?

**Comment by statik** [comment]
Thank you @aiuto! Life is busy, thank you for your work maintaining rules_pkg.

**Comment by aiuto** [comment]
To answer @Gloix 's question. This looks like it will do all files in the same way.
I think of this as a stop-gap.  The more principled solution will use pkg_filegroup (when that gets finished) to finely specify modes and owners (in platform agnostic way) on a per-file basis.

**Pull request merged**